app/main.py

```diff
@@ -1 +1,33 @@
-# write your code here
+import json
+from decimal import Decimal
+
+
+def calculate_profit(file_name):
+    with open(file_name, "r") as file_trades:
+        trades = json.load(file_trades)
+
+    earned_money = 0
+    matecoin_account = 0
+
+    for trade in trades:
+        matecoin_price = Decimal(trade["matecoin_price"])
+
+        if trade["bought"]:
+            earned_money -= Decimal(trade["bought"]) * matecoin_price
+            matecoin_account += Decimal(trade["bought"])
+
+        elif trade["sold"]:
```

Comment:
It would be better to use `if` here instead of `elif`. Because theoretically there can be both 'sold' and 'bought'.